pyanaconda/modules/subscription/subscription.py

```diff
@@ -242,5 +441,438 @@ def set_system_purpose_with_task(self, sysroot):
+
+        # if baseurl is non-empty, set it to RHSM
+        if baseurl:
+            if self._original_rhsm_baseurl_set:
+                # save the original CDN baseurl
+                self._original_rhsm_baseurl = self.rhsm_config_proxy.Get("rhsm.baseurl", "")
+            # set the custom CDN baseurl to RHSM
+            self.rhsm_config_proxy.Set("rhsm.baseurl", get_variant(Str, baseurl), "")
+            self._original_rhsm_baseurl_set = False
+            log.debug("Red Hat CDB baseurl set to: %s", baseurl)
+        else:
+            # if baseurl is empty, restore original value
+            self.rhsm_config_proxy.Set("rhsm.baseurl", get_variant(Str, self._original_rhsm_baseurl), "")
+            self._original_rhsm_baseurl_set = True
+            log.debug("Red Hat CDN baseurl set back to: %s", self._original_rhsm_baseurl)
+
+        self.rhsm_baseurl_changed.emit()
+
+    @property
+    def server_proxy_hostname(self):
+        """Get the hostname of the RHSM HTTP proxy.
+
+        Empty string means that no proxy hostname has been set.
+
+        :returns: RHSM HTTP proxy hostname
+        :rtype: str
+        """
+        return self._server_proxy_hostname
+
+    @property
+    def server_proxy_port(self):
+        """Get the port of the RHSM HTTP proxy.
+
+        -1 means no proxy port has been set.
+
+        :returns: RHSM HTTP proxy port
+        :rtype: int
+        """
+        return self._server_proxy_port
+
+    @property
+    def server_proxy_user(self):
+        """Get the access username for the RHSM HTTP proxy.
+
+        Empty string means that no username has been set.
+
+        :returns: RHSM HTTP proxy access username
+        :rtype: str
+        """
+        return self._server_proxy_user
+
+    @property
+    def server_proxy_password_set(self):
+        """Report if the access password for the RHSM HTTP proxy has been set.
+
+        :return: if the access password for RHSM HTTP proxy has been set
+        :rtype: bool
+        """
+        return bool(self._server_proxy_password)
+
+    # pylint: disable=too-many-function-args
+    def set_server_proxy(self, hostname, port, username, password):
+        """Set configuration of the RHSM HTTP proxy.
+
+        :param str hostname: RHSM HTTP proxy hostname
+        :param int post: RHSM HTTP proxy port (set to -1 to clear)
+        :param str username: RHSM HTTP proxy access username
+        :param str password: RHSM HTTP proxy access password
+        """
+        # set the local attributes
+        self._server_proxy_hostname = hostname
+        self._server_proxy_port = port
+        self._server_proxy_user = username
+        self._server_proxy_password = password
+        # negative port number means no port has been set
+        if port < 0:
+            port_string = ""
+        else:
+            port_string = "{}".format(port)
+        # set data to RHSM
+        self.rhsm_config_proxy.Set("server.proxy_hostname", get_variant(Str, hostname), "")
```

Comment:
The last parameter is for setting locale for RHSM output (mainly for error messages I think). While we do use this in some of the installation tasks, as the results are user visible, I this case I think we should keep the default in place. Otherwise, RHSM error messages in logs would be translated to currently selected language and potentially unreadable as a result.